### ISSUE TITLE
ajaxSubmit is not encoding newlines as CRLF pairs, but only as LF

### ISSUE THREAD

**Issue by hchonov** [jquery-form/form#517]
@see pull request #516

**Comment by kevindb** [comment]
This looks related to #158. How do you compare your fix vs @cgi-caesar's? Should we implement _both_ of your changes, or would that break things further?

**Comment by hchonov** [comment]
I think that it makes sense of using only the jQuery method for serializing arrays as it is doing other stuffs as well and if there are changes later to it we will get automatically an updated version and only jQuery will be responsible for the encoding and not jQuery-form anymore. 

I don't think that for this case we need both issues, one of those should be sufficient, but I would prefer using the serializeArray jQuery method and letting all the responsibility to jQuery.

Implementing both solutions should not break things further, but it will add additional overhead.

I would say it is up to the maintainer to decide, which is better :).

**Comment by kevindb** [comment]
While I agree with you that using jQuery's serializeArray() potentially makes sense in the long term, it would fundamentally change how the form plugin is serializing the form today.
In your PR #516, you have the line

```javascript
a = this.serializeArray(this.formToArray(options.semantic, elements, options.filtering));
```

[jQuery.serializeArray()](https://api.jquery.com/serializeArray/) takes no arguments, so I think the result of this.formToArray() is being thrown away in favor of jQuery's serialization. This means that the `semantic` and `filtering` options would not have any effect on the resulting serialized form values.

I have merged a modified version of @cgi-caesar's PR #421 that is consistent with [jQuery's serialization source](https://github.com/jquery/jquery/blob/ac9e3016/src/serialize.js#L124).
Please let me know if this change resolves your issue.

**Comment by kevindb** [comment]
[v4.1.0](https://github.com/jquery-form/form/releases/tag/v4.1.0) has been released with this change.
